src/main/java/com/yelp/nrtsearch/server/grpc/LuceneServer.java

```diff
@@ -775,7 +787,11 @@ public void commit(
             () -> {
               try {
                 IndexState indexState = globalState.getIndex(commitRequest.getIndexName());
-                long gen = indexState.commit();
+                long gen =
+                    indexState.commit(
+                        commitRequest.getDisableV0Archiver()
```

Comment:
We can have a parameter in `indexState.commit` called `uploadToS3IfEnabled`. We can pass that true from here and false from BackupHandler

Reply:
as discussed in person uses the `backupWithIncArchiver` flag here